internal/backend/runtime/omni/controllers/omni/config_patch_metrics.go

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2025 Sidero Labs, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+
+package omni
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/cosi-project/runtime/pkg/controller"
+	"github.com/cosi-project/runtime/pkg/safe"
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/zap"
+
+	"github.com/siderolabs/omni/client/pkg/omni/resources"
+	"github.com/siderolabs/omni/client/pkg/omni/resources/omni"
+)
+
+// ConfigPatchMetricsController publishes metrics on config patches.
+//
+//nolint:govet
+type ConfigPatchMetricsController struct {
+	metricsOnce sync.Once
+	patchSizes  *prometheus.HistogramVec
+}
+
+// Name implements controller.Controller interface.
+func (ctrl *ConfigPatchMetricsController) Name() string {
+	return "ConfigPatchMetricsController"
+}
+
+// Inputs implements controller.Controller interface.
+func (ctrl *ConfigPatchMetricsController) Inputs() []controller.Input {
+	return []controller.Input{
+		{
+			Namespace: resources.DefaultNamespace,
+			Type:      omni.ConfigPatchType,
+			Kind:      controller.InputWeak,
+		},
+	}
+}
+
+// Outputs implements controller.Controller interface.
+func (ctrl *ConfigPatchMetricsController) Outputs() []controller.Output {
+	return nil
+}
+
+func (ctrl *ConfigPatchMetricsController) initMetrics() {
+	ctrl.metricsOnce.Do(func() {
+		ctrl.patchSizes = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+			Name: "omni_machine_config_patch_size",
```

Comment:
```suggestion
			Name: "omni_config_patch_size",
```

This is not specifically any type of a config patch, but rather all config patches.

Reply:
👍